index.rb

```diff
@@ -0,0 +1,180 @@
+module Enumerable # rubocop:disable Metrics/ModuleLength
+  def my_each
+    if block_given?
+      0.upto(length - 1) do |i|
+        yield self[i]
+      end
+    else
+      puts 'no block is given'
+    end
+    self
+  end
+
+  def my_each_with_index
+    if block_given?
+      0.upto(length - 1) do |i|
+        yield(self[i], i)
+      end
+    else
+      puts 'no block is given'
+    end
+    self
+  end
+
+  def my_select
+    array = []
+    if block_given?
+      0.upto(length - 1) do |i|
+        array << self[i] if yield self[i]
+      end
+    else
+      puts 'no block is given'
+    end
+    array
+  end
+
+  def my_all
+    value = true
+    if block_given?
+      0.upto(length - 1) do |i|
+        value = false unless yield self[i]
+      end
+    else
+      puts 'no block is given'
+    end
+    value
+  end
```

Comment:
#### It is important to note that `my_any?`, `my_all?` and `my_none?` can accept an argument and can also be called without any argument or block. See some cases the current method does not cover below:

 - When a pattern is passed this method should return true if every collection member matches this pattern. Note that pattern could be a `Class` (e.g. `Integer`, `String`) or `RegEx`
   ```ruby
   # When a Class pattern is passed
   irb(main):014:0> [1,2,3].all?(Integer)
   => true
   irb(main):015:0> [1,2,3].my_all(Integer)
   => ArgumentError (wrong number of arguments (given 1, expected 0))

   # When a RegEx pattern is passed
   irb(main):018:0> %w[ant bat cat].all?(/t/)
   => true
   irb(main):019:0> %w[ant bat cat].my_all(/t/)
   => ArgumentError (wrong number of arguments (given 1, expected 0))

   # When no argument is passed
   irb(main):013:0> [nil, true, false].all?
   => false
   irb(main):014:0> [nil, true, false].my_all
   no block is given
   => true
   ```
   Hint: Use ` === ` to check if a member matches a patern.

### Find more details about [`all?` here](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F)

### Please let's do this for `my_any?` and `my_none?` too.
_Also, note that methods that return `true` or `false` should end with `?`_